### PR TITLE
Update release script

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
@sosukesuzuki Do you want review this before you do next release?

**Comment by sosukesuzuki** [comment]
@fisker I'm sorry, I'll review it after releasing 2.2

File: scripts/release/package.json [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+    "minimist": "1.2.5",
     "node-fetch": "2.6.1",
     "outdent": "0.7.1",
-    "semver": "5.5.0",
-    "string-width": "2.1.1"
-  }
+    "semver": "7.3.2",
+    "string-width": "4.2.0"
+  },
+  "version": "3.0.0"
```
Comment (sosukesuzuki) [inline scripts/release/package.json:12]: question, why did you add `version` ?

File: scripts/release/release.js [hunk 2]
```diff
@@ -2,25 +2,29 @@
...
-  const chalk = require("chalk");
-  const { string: outdentString } = require("outdent");
-  const minimist = require("minimist");
-  const semver = require("semver");
-
-  const { readJson } = require("./utils");
+  // Fetch git tags to get the previous version number (i.e. the latest tag)
+  await runGit(["fetch", "--tags"], { cwd: __dirname });
+  // Install script's dependencies before any require
+  await runYarn(["install"], { cwd: __dirname });
```
Comment (sosukesuzuki) [inline scripts/release/release.js:15]: if dependencies of `./scripts/release` isn't installed yet, this script is failed at line:9 via requiring `./utils.js`. So I think it should be executed before executing any `require`.

File: scripts/release/utils.js [hunk 3]
```diff
@@ -19,27 +20,36 @@ function fitTerminal(input) {
   return input;
 }
 
-function logPromise(name, promise) {
+async function logPromise(name, promise) {
+  if (typeof promise === "function") {
+    promise = promise();
+  }
```
Comment (sosukesuzuki) [inline scripts/release/utils.js:26]: This variable name is hard to understand to me. So what do you think about the below suggestion?

```suggestion
async function logPromise(name, promiseOrAsyncFunction) {
  const promise =
    typeof promiseOrAsyncFunction === "function"
      ? promise()
      : promiseOrAsyncFunction;
```

**Review by sosukesuzuki — COMMENTED** [review]
Sorry for delay reviewing.

File: scripts/release/package.json (hunk 1 above)
Comment (fisker) [reply scripts/release/package.json]: I forgot... Removed https://github.com/prettier/prettier/pull/9418/commits/7ed0cca9cef4b42685a95321b63360403fe77148.

File: scripts/release/release.js (hunk 2 above)
Comment (fisker) [reply scripts/release/release.js]: Fixed https://github.com/prettier/prettier/pull/9418/commits/82c0c83bee3dd9bae3b836f5320f6aa475ddc0fe

**Comment by thorn0** [comment]
Are we sure this is safe to merge? ;)

**Comment by fisker** [comment]
I hope so, but if there is problem, you can fix during release.
